### ISSUE TITLE
Feature request: vectorized character slice

### ISSUE THREAD

**Issue by mwtoews** [numpy/numpy#8579]
Inspecting `numpy.core.defchararray` and [the online docs for string operations](https://docs.scipy.org/doc/numpy/reference/routines.char.html), it seems there is a missing feature to slice a string array in a vectorized way.

It would be nice to have `numpy.core.defchararray.slice(a, start, stop[, step])`

[This Q/A](http://stackoverflow.com/q/39042214/327026) seems to reflect this missing feature, with a few example implementations.

**Comment by shoyer** [comment]
Yes, this seems like a very straightforward and obviously useful addition.

I encourage you to put together a pull request.
